feat(correlation): use people_url to load people modal

hazzadous opened the pull request:
This change adds a new method to the `personsModelLogic` to allow for
specifying a url to be used to retrieve people from. I've avoided
overloading the existing `loadPeople` and rather added a new one,
`lostPeopleFromUrl`.

Currenlty this is only being used by correlations, but the intention is
to extend it's usage to anywhere else that is performing drilldowns to
people from, for instance, the trends aggregation.

Commit:
feat(correlation): use people_url to load people modal

This change adds a new method to the `personsModelLogic` to allow for
specifying a url to be used to retrieve people from. I've avoided
overloading the existing `loadPeople` and rather added a new one,
`lostPeopleFromUrl`.

Currenlty this is only being used by correlations, but the intention is
to extend it's usage to anywhere else that is performing drilldowns to
people from, for instance, the trends aggregation.

Commit:
add paycard back

Commit:
make labels look right

Commit:
remove unused code

Commit:
fix typing in test

Commit:
fix typing in personsModelLogic

Commit:
refactor to make eventUsageLogic work

Commit:
fix typing

Commit:
fix typing again

EDsCODE commented on frontend/src/scenes/trends/personsModalLogic.tsx:
```diff
@@ -262,6 +272,20 @@ export const personsModalLogic = kea<personsModalLogicType<PersonModalParams>>({
                 return peopleResult
             },
+            loadPeopleFromUrl: async ({ url, stepNumber, breakdown, breakdown_value, label }) => {
+                const people = await (await fetch(url)).json()
```
`await api.get(...)` would be simpler/consistent here

hazzadous replied:
Have updated

EDsCODE commented on frontend/src/scenes/trends/personsModalLogic.tsx:
```diff
@@ -262,6 +272,20 @@ export const personsModalLogic = kea<personsModalLogicType<PersonModalParams>>({
 
                 return peopleResult
             },
+            loadPeopleFromUrl: async ({ url, stepNumber, breakdown, breakdown_value, label }) => {
```
Do you think we could parse some of this directly from the URL itself? That would reduce this function to just requiring a `url`

hazzadous replied:
Yeah was thinking about that. These additional args are specific things for generating the label for display. I wouldn't want to parse the url, if smells. I don't want to start coupling things like this to what can be, at the moment, be treated as an opaque url.

If we really care about making sure that the label consistent with the data returned, we can either pass along an object along with the url, containing the info (could be the exact label although that seems a little prescriptive for what feels contextual to the consumer). Or we can return this data within the response?

EDsCODE commented on frontend/src/scenes/funnels/funnelLogic.ts:
```diff
@@ -1129,19 +1128,42 @@ export const funnelLogic = kea<funnelLogicType>({
-                label: entity.id,
-                date_from: '',
-                date_to: '',
-                filters: {
```
great to see all this removed

EDsCODE reviewed (approved):
Some questions on making some bits even simpler. Otherwise lgtm

Commit:
Merge remote-tracking branch 'origin/master' into gh-6953-frontend

hazzadous commented:
Just merging in master to see if it prompts the e2e tests to pass

neilkakkar commented on frontend/src/scenes/funnels/funnelLogic.ts:
```diff
@@ -1129,19 +1128,42 @@ export const funnelLogic = kea<funnelLogicType>({
+                personsModalLogic.actions.loadPeopleFromUrl({
+                    url: success ? correlation.success_people_url : correlation.failure_people_url,
+                    // I'm not sure why this is -2, I'm just copying from what was here previously
+                    stepNumber: success ? values.stepsWithCount.length : -2,
```
don't need this anymore, it's already in the url now? (was https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/funnels/funnelLogic.ts#L1129 funnelStep, which we've overridden with `funnel_custom_steps` .

hazzadous replied:
These additional vars I kept as I thought they are needed for setting the title of the modal. I'll remove if not @neilkakkar 


If they are still needed, I'd rather keep this PR small. We could follow up with an addition to the people endpoint to include the specific filtering detail that could then be used for display, or have this to hand in an easy way in the initial response?

I'll add a NOTE about why it's still needed, unless you think we can remove them @neilkakkar

neilkakkar replied:
Ah right, this tells whether it's success or failure? Cool cool

neilkakkar commented on frontend/src/scenes/funnels/funnelLogic.ts:
```diff
@@ -1129,19 +1128,42 @@ export const funnelLogic = kea<funnelLogicType>({
+                    url: success ? correlation.success_people_url : correlation.failure_people_url,
+                    stepNumber: success ? values.stepsWithCount.length : -2,
+                    label: name,
+                    breakdown: '',
```
StepNumber, and breakdown don't make sense here anymore?

hazzadous replied:
I think stepNumber is still needed to have the right modal title, although I'm just about to test this.

The breakdown and breakdown_value yup I guess shouldn't need to set that here, I'll make them optional.

hazzadous replied:
Simplified a little, but yeah, need this for display.

Can clean up in another PR

Commit:
remove some funnel stories

Commit:
get storybook working with paycard

Commit:
cleaned up the interface a little

Commit:
Merge remote-tracking branch 'origin/master' into gh-6953-frontend

Commit:
chore(correlation): use `api.get` for fetching people

This updates to use api.get, and also converts the returned uri from the
correlation endpoint to an absolut uri, instead of just an absolute
path.

The `api.get` method expects either an abolute uri, or a relative path.
I'd rather not use an relative path as it's not obvious what it should
be relative to.

Commit:
fix tests

Commit:
make storybook uris absolute
